napari/_qt/layer_controls/qt_image_controls_base.py

```diff
@@ -193,9 +193,19 @@ def mouseMoveEvent(self, event):
         self.layer.status = self.layer._contrast_limits_msg
 
     def closeEvent(self, event):
-        self.deleteLater()
+        self.close()
         event.accept()
 
+    def close(self):
+        """Layer widget is closing."""
+        super().close()
+        self.layer.events.colormap.disconnect(self._on_colormap_change)
```

Comment:
Do these correspond 1-1 to events that get connected in the `__init__` constructor? I'm wondering if we can make this more easily discoverable and explicit, for example by creating twin `self._events_connect` and `self._events_disconnect` that are defined together in the code?

Reply:
I'm open to that. I think the more explicit we are about connecting and disconnecting events the better